platforms/sdl/desktop/AppPlatform_sdl.cpp

```diff
@@ -81,6 +81,8 @@ static int save_png(const char *filename, unsigned char *pixels, int line_size,
 	}
 	if (rows != NULL)
 	{
+        // Somehow this is erroring with the following on Mac OS X: pointer being freed was not allocated
+        // I'm not changing this because I'm quite certain it's correct, and doing the old implementation would certainly leak memory
```

Comment:
I think I encountered a similar error on Linux too. I wasn't able to reproduce it though.

Reply:
I didn't even notice! This is completely wrong.

`rows` is an array of `png_bytep` (so basically a `uint8_t**`). So you should be using **`delete[] rows;`** instead of `delete rows[height];`.

1. You are deleting a row, which is not allowed.
2. You're even deleting out of bounds, which is BAD.

Reply:
As someone who's knowledge of C++ is... patchy, I thought that was some special-C++ `delete[]` syntax, similar to how C arrays are created like `int x[5]`.

The LibPNG code in general is a bit of a mess. My Android PR switches over to `stb_image` (like literally every other platform in this project) and it's much simpler.

Reply:
No. `delete x;` deletes the object `x`, and `delete[] y;` deletes the array of objects `y`. In this case, `y` actually stores some hidden metadata (which is implementation specific) which stores the amount of objects that the array actually manages, so that it can easily destruct all the objects.

Reply:
Fixed this. Thanks iProgram for showing us the light.